### PR TITLE
Parameter for virtual battery in SITL

### PR DESCRIPTION
**Pull request by potaito**
Adds a parameter to configure the battery level in SITL. Useful for testing on different battery levels. From a PX4 developer perspective this is probably not necessary, because it's easy and quick to change the hard-coded value when necessary. But for people creating their own apps using the DroneCode SDK, this might be useful.

### CODE REVIEW

**Comment by julianoes** [comment]
Would be nice to have a setter for MAVSDK for this param.

**Comment by potaito** [comment]
> Would be nice to have a setter for MAVSDK for this param.

Do I have to add something for that?

**Comment by julianoes** [comment]
Oh I think I misunderstood that. It's just a SITL config, so a param to set using `param set` or QGC is probably fine.

File: src/modules/simulator/simulator_params.c [hunk 1]
```diff
@@ -51,3 +51,15 @@
  * @group SITL
  */
 PARAM_DEFINE_FLOAT(SIM_BAT_DRAIN, 60);
+
+/**
+ * Simulator Battery minimal percentage
```
Comment (bkueng) [inline src/modules/simulator/simulator_params.c]: Can you extend the description for when and how to change this?

File: src/modules/simulator/simulator_mavlink.cpp [hunk 2]
```diff
@@ -383,12 +383,11 @@ void Simulator::handle_message_hil_sensor(const mavlink_message_t *msg)
 		}
 
 		float ibatt = -1.0f; // no current sensor in simulation
-		const float minimum_percentage = 0.499f; // change this value if you want to simulate low battery reaction
 
 		/* Simulate the voltage of a linearly draining battery but stop at the minimum percentage */
 		float battery_percentage = 1.0f - (now_us - batt_sim_start) / discharge_interval_us;
 
-		battery_percentage = math::max(battery_percentage, minimum_percentage);
+		battery_percentage = math::max(battery_percentage, _battery_min_percentage.get() / 100);
```
Comment (bkueng) [inline src/modules/simulator/simulator_mavlink.cpp]:
```suggestion
		battery_percentage = math::max(battery_percentage, _battery_min_percentage.get() / 100.f);
```

File: src/modules/simulator/simulator_params.c (hunk 1 above)
Comment (potaito) [reply src/modules/simulator/simulator_params.c]: Done :+1:

**Comment by potaito** [comment]
Thanks for the review @bkueng

**Pull request merged**